WcaOnRails/app/models/competition.rb

```diff
@@ -768,7 +768,12 @@ def guests_entry_fee_required?
   end
 
   def registration_period_required?
-    use_wca_registration? || (confirmed? && created_at.present? && created_at > Date.new(2018, 9, 13))
+    (
+      use_wca_registration? || (confirmed? && created_at.present? && created_at > Date.new(2018, 9, 13)) ||
+
+      # This is to avoid situations like the one described in https://github.com/thewca/worldcubeassociation.org/issues/4257
+      on_the_spot_entry_fee_required?
```

Comment:
It felt kind of dirty to me to update the view to handle this, but honestly I'm not sure of what's better in this case.

However, `on_the_spot_entry_fee_lowest_denomination` can't be set without checking `on_the_spot_registration`, which automatically makes `on_the_spot_entry_fee_lowest_denomination` required, thus making the registration period required too with this approach.

Reply:
What I meant was that if `on_the_spot_registration` is set, but there is no country (which is not required until confirmed/visible) then this returns false: https://github.com/thewca/worldcubeassociation.org/blob/f71a6178ea16315b1b1d33dce50fdf71e2fb4748/WcaOnRails/app/models/competition.rb#L740-L748
In that case registration period wouldn't be required, but you could still set `on_the_spot_entry_fee_lowest_denomination`. I might've gotten this wrong though!

Reply:
I think I like the idea of updating the view code to handle this, instead of making our validations weirder/stricter. I think it's fairly straightforward to update the view code to check for `@competition.registration_close.nil?` and do... something. As Jonatan pointed out above, since this only matters for unconfirmed competitions, it doesn't even matter that much what we say (it could be something like "Registration end date not yet set").

Reply:
Ah, right, I forgot about that country hack we had to introduce! Ok, I'll update the view then.